rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -2,5 +2,1349 @@
+    pub version: String,
+    /// The namespace of the message schema.
+    pub registry_namespace: String,
+}
+
+impl From<AssetStatus> for adr_name_gen::AssetStatus {
+    fn from(value: AssetStatus) -> Self {
+        adr_name_gen::AssetStatus {
+            config: value.config.map(StatusConfig::into),
+            datasets: option_vec_from(value.datasets_schema, AssetDatasetEventStreamStatus::into),
+            events: option_vec_from(value.events_schema, AssetDatasetEventStreamStatus::into),
+            management_groups: option_vec_from(
+                value.management_groups,
+                AssetManagementGroupStatus::into,
+            ),
+            streams: option_vec_from(value.streams, AssetDatasetEventStreamStatus::into),
+        }
+    }
+}
+
+impl From<AssetDatasetEventStreamStatus> for adr_name_gen::AssetDatasetEventStreamStatus {
+    fn from(value: AssetDatasetEventStreamStatus) -> Self {
+        adr_name_gen::AssetDatasetEventStreamStatus {
+            name: value.name,
+            message_schema_reference: value
+                .message_schema_reference
+                .map(MessageSchemaReference::into),
+            error: value.error.map(ConfigError::into),
+        }
+    }
+}
+
+impl From<AssetManagementGroupStatus>
+    for adr_name_gen::AssetManagementGroupStatusSchemaElementSchema
+{
+    fn from(value: AssetManagementGroupStatus) -> Self {
+        adr_name_gen::AssetManagementGroupStatusSchemaElementSchema {
+            actions: option_vec_from(value.actions, AssetManagementGroupActionStatus::into),
+            name: value.name,
+        }
+    }
+}
+
+impl From<AssetManagementGroupActionStatus>
+    for adr_name_gen::AssetManagementGroupActionStatusSchemaElementSchema
+{
+    fn from(value: AssetManagementGroupActionStatus) -> Self {
+        adr_name_gen::AssetManagementGroupActionStatusSchemaElementSchema {
+            error: value.error.map(ConfigError::into),
+            name: value.name,
+            request_message_schema_reference: value
+                .request_message_schema_reference
+                .map(MessageSchemaReference::into),
+            response_message_schema_reference: value
+                .response_message_schema_reference
+                .map(MessageSchemaReference::into),
+        }
+    }
+}
+
+impl From<MessageSchemaReference> for adr_name_gen::MessageSchemaReference {
+    fn from(value: MessageSchemaReference) -> Self {
+        adr_name_gen::MessageSchemaReference {
+            schema_name: value.name,
+            schema_version: value.version,
+            schema_registry_namespace: value.registry_namespace,
+        }
+    }
+}
+
+// ~~~~~~~~~~~~~~~~~~~DTDL Equivalent Enums~~~~~~~
+// TODO: remove in favor of Rust enum
+/// The target of the event or stream.
+#[derive(Clone, Debug)]
+pub enum EventStreamTarget {
+    /// MQTT
+    Mqtt,
+    /// Storage
+    Storage,
+}
+
+/// The MQTT `QoS` setting.
+#[derive(Clone, Debug)]
+pub enum Qos {
```

Comment:
For the future, could we leverage the struct we already have for the mqtt crate?

Reply:
So having a From<QoS> for azure_iot_operations_mqtt::QoS (not sure if this is the actual name)